src/lang/codeManager.ts

```diff
@@ -165,13 +165,16 @@ export default class CodeManager {
     }
   }
 
-  async updateEditorWithAstAndWriteToFile(ast: Program) {
+  async updateEditorWithAstAndWriteToFile(
+    ast: Program,
+    options?: Partial<{ isDeleting: boolean }>
+  ) {
     // We clear the AST when it cannot be parsed. If we are trying to write an
     // empty AST, it's probably because of an earlier error. That's a bad state
     // to be in, and it's not going to be pretty, but at the least, let's not
-    // permanently delete the user's code. If you want to clear the scene, call
-    // updateCodeStateEditor directly.
-    if (ast.body.length === 0) return
+    // permanently delete the user's code accidentally.
+    // if you want to clear the scene, pass in the `isDeleting` option.
+    if (ast.body.length === 0 && options?.isDeleting === false) return
```

Comment:
If the option isn't provided, this will be `undefined`. The feature tree should need to call this with `isDeleting: true`. Until that change is made, this PR shouldn't be merged.

```suggestion
    if (ast.body.length === 0 && !options?.isDeleting) return
```

Reply:
Doh here I thought I was being clever. Thank you!